src/fight/mod.rs

```diff
@@ -3,156 +3,301 @@
 ///
-/// # Parameters
-/// nb_wounds (usize): The number of wounds the defending Regiment will suffer
+/// ### Parameters
+/// (&regiment::Regiment) attacking_regiment -> The attacker
 ///
-/// regiment_defending_armor (usize): The armor of the defending Regiment
+/// (&regiment::Regiment) defending_regiment -> The defender
 ///
-/// regiment_attacking_armor_penetration (usize): The armor penetration of the attacking Regiment
+/// (ComputeCase) case -> The scenario from first_unit point of view
 ///
-/// # Return
-/// usize: The number of wounds the defending Regiment will suffer after armor save
-fn get_final_result(
-    nb_wounds: usize,
-    regiment_defending_armor: usize,
-    regiment_attacking_armor_penetration: usize,
-) -> usize {
-    let to_save: i8 = regiment_defending_armor as i8 - regiment_attacking_armor_penetration as i8;
-    let throw_to_save: i8 = match to_save {
-        i8::MIN..=0 => return nb_wounds,
-        1 => 6,
-        2 => 5,
-        3 => 4,
-        4 => 3,
-        5..=i8::MAX => 2,
+/// ### Return
+/// (usize, f64) -> The average amount of damage dealt by first_unit and the probability for this scenario to occurs
+fn compute_case(
+    attacking_regiment: &regiment::Regiment,
+    defending_regiment: &regiment::Regiment,
+    case: ComputeCase,
+) -> (usize, f64) {
+    let attacking_model: &model::Model = attacking_regiment.get_model();
+    let defending_model: &model::Model = defending_regiment.get_model();
+    let nb_touch: usize = (attacking_model.get_stats().get_attack() as f64
+        * 1.5
+        * attacking_regiment.get_cols() as f64)
+        .round() as usize;
+    let touch_probability: f64 =
+        computation_tools::compute_damage_probability(attacking_model, defending_model);
+    let mut results: Vec<(usize, f64)> = Vec::new();
+    let mut selection: Vec<(usize, f64)> = Vec::new();
+    let mut bondaries: (isize, isize);
+    let threshold: f64 = match case {
+        ComputeCase::BEST => global_values::BEST_CASE_THRESHOLD,
+        ComputeCase::WORST => global_values::WORST_CASE_THRESHOLD,
+        _ => 0.0_f64,
+    };
+    let defender_hp = defending_model.get_stats().get_health_point();
+
+    if let ComputeCase::MEAN = case {
+        return compute_mean_case(attacking_regiment, defending_regiment);
+    }
+    for i in 0..nb_touch {
+        bondaries = match case {
+            ComputeCase::BEST => (i as isize, nb_touch as isize),
+            ComputeCase::WORST => (0, i as isize),
+            ComputeCase::MEAN => unreachable!("Code not supposed to be reached!"),
+        };
+        results.push((
+            i,
+            math_tools::sigma(
+                bondaries.0,
+                bondaries.1,
+                |curr, param, _, _| {
+                    math_tools::compute_bernoulli(
+                        param.unwrap().1 as usize,
+                        curr as usize,
+                        param.unwrap().0,
```

Comment:
Error is impossible as param is an optional parameter passed to the function executed by sigma()
it worth None if no parameter has been passed or Some(x) otherwise.
And I gave a parameter like 3lines just under so it can't be None